### PR TITLE
WT-3264 Fix log file cleanup when toggling logging.

### PR DESCRIPTION
**Pull request by sueloverso**
Not yet ready for review.

### CODE REVIEW

**Comment by sueloverso** [comment]
@michaelcahill and @agorrod this change is ready for review.  With it WT will detect logging getting toggled on back and forth and if the log files have been removed it will start logging at a log file larger than the largest checkpoint LSN we find in the meta-data.  I have a change that I'm preparing for mongoDB to remove all the log files if they're turning logging off so that we don't continually do this toggling dance forever.

File: src/txn/txn_recover.c [hunk 1]
```diff
@@ -443,9 +449,24 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
...
+		 */
 		WT_ERR(__recovery_file_scan(&r));
 		conn->next_file_id = r.max_fileid;
-		goto done;
+
+		if (FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) &&
+		    WT_IS_MAX_LSN(&metafile->ckpt_lsn) &&
+		    !WT_IS_MAX_LSN(&r.max_lsn)) {
+			WT_ERR(__wt_log_reset(session, r.max_lsn.l.file));
+			goto ckp;
```
Comment (agorrod) [inline src/txn/txn_recover.c]: nit: could you make this ``ckpt``?

Comment (sueloverso) [inline src/txn/txn_recover.c]: Just to make sure I'm interpreting that correctly, do you mean you'd like `max_lsn` to be `max_ckpt_lsn`?

Comment (sueloverso) [inline src/txn/txn_recover.c]: Oh, I think you meant `goto ckp` to be `goto ckpt`.

Comment (agorrod) [inline src/txn/txn_recover.c]: It was the ``goto``

**Pull request merged**